modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <div 
+    v-if="guest"
+    class="admin-bar-guest"
+  >
+    <button @click="logout">Log Out</button>
+  </div>
+  <div v-else
+    data-apos-test="adminBar"
+    class="apos-admin-bar-wrapper"
+    :class="themeClass"
+  >
+    <div ref="spacer" class="apos-admin-bar-spacer" />
+    <nav ref="adminBar" class="apos-admin-bar">
+      <div class="apos-admin-bar__row">
+        <AposLogoPadless class="apos-admin-bar__logo" />
+        <TheAposAdminBarMenu :items="menuItems" />
+        <TheAposAdminBarLocale v-if="hasLocales()" />
+        <TheAposAdminBarUser
+          data-apos-test="authenticatedUserMenuTrigger"
+          class="apos-admin-bar__user"
+          :items="userItems"
+        />
+      </div>
+      <TheAposContextBar @visibility-changed="setSpacer" />
+      <component
+        v-bind="bar.props || {}"
+        :is="bar.componentName"
+        v-for="bar in bars"
+        :key="bar.id"
+      />
+    </nav>
+  </div>
+</template>
+
+<script>
+import AposThemeMixin from 'Modules/@apostrophecms/ui/mixins/AposThemeMixin';
+
+export default {
+  name: 'TheAposAdminBar',
+  mixins: [ AposThemeMixin ],
+  props: {
+    items: {
+      type: Array,
+      required: true
+    }
+  },
+  computed: {
+    menuItems() {
+      return this.items.filter(item => !item.options?.user);
+    },
+    userItems() {
+      return this.items.filter(item => item.options?.user);
+    },
+    moduleOptions() {
+      return window.apos.adminBar;
+    },
+    guest() {
```

Comment:
Exposes the fact that the user is a guest. See also the `getBrowserData` extension below.